cgi-bin/DW/Controller/Auth.pm

```diff
@@ -52,12 +52,11 @@ sub logout_handler {
         }
 
         # If the logout form asked to be sent back to the original page (with a
-        # hidden 'ret=1' form input), do so (as long as the logout was
-        # successful).
-        if ( $vars->{success} && $post_args->{ret} ) {
-            my $referer = $r->header_in('Referer');
-            if ( $referer && LJ::check_referer( '', $referer ) ) {
-                return $r->redirect($referer);
+        # hidden 'ret=1' form input and a return url), do so (as long as the logout
+        # was successful).
+        if ( $vars->{success} && $post_args->{returnto} && $post_args->{ret} ) {
```

Comment:
After some discussion in the discord, we've decided to keep the `ret` option as a way to override on site pages we don't want to return back to after logout (because they require being logged in, etc).